### PR TITLE
Update limited medical authorizations in the 21-22 and 21-22a form generation #98295

### PR DESCRIPTION
**Pull request by opticbob**
## Summary

- *This work is behind a feature toggle (flipper): YES*
- *(Summarize the changes that have been made to the platform):* This PR changes how we represent the choices for limitation of consent on the 2122 and 2122a PDFs.  For the 2122 the choices have been inverted to match the legal language on the form.  On the 2122a language has been added to make it more clear what the limitations are for.
- *(If bug, how to reproduce)*
- *(What is the solution, why is this the solution?):* The 2122 changes were necessary to match the legal meaning of the form.
- *(Which team do you work for, does your team own the maintenance of this component?):* FAR/ARM
- *(If introducing a flipper, what is the success criteria being targeted?)*

## Related issue(s)

- https://github.com/department-of-veterans-affairs/va.gov-team/issues/98295
- *Link to previous change of the code/bug (if applicable)*
- *Link to epic if not included in ticket*

## Testing done

- [x] *New code is covered by unit tests*
- *Describe what the old behavior was prior to the change:* The 2122 PDF had the same limitations checked as input in the web app, when they should have been the opposite.  That is now resolved.  The 2122a limitations sentence now has a clear beginning and a period at the end.
- *Describe the steps required to verify your changes are working as expected. Exclusively stating 'Specs run' is NOT acceptable as appropriate testing:* Use the staging frontend to create a data request body, then submit it to the backend.  You should get PDFs generated with the appropriate limitations filled out.
- *If this work is behind a flipper:*
  - *Tests need to be written for both the flipper on and flipper off scenarios. [Docs](https://depo-platform-documentation.scrollhelp.site/developer-docs/feature-toggles-guide#Featuretogglesguide-Backendexample).*
  - *What is the testing plan for rolling out the feature?*

## Screenshots
_Note: Optional_

## What areas of the site does it impact?
*(Describe what parts of the site are impacted and*if*code touched other areas):* Just the representation management module.

## Acceptance criteria

- [x]  I updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [ ]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

## Requested Feedback

This PR has a couple of items of note.  There are changes to the factories for `accredited_individual` and `accredited_organization`.  These are in place to make generating the PDF fixture file updates in the future easier.  The second thing is that we now have code in place to produce unflattened PDFs that we use for testing but aren't ever exposed to the user.  When trying to test the checkbox changes I realized that our current tests only detected text changes.  I got a tip on a helper that would test all the fields to make sure they matched but it required unflattened PDFs to work.  With the new tests in place, any checkbox behavior change should get caught by the tests.

### CODE REVIEW

File: modules/representation_management/lib/representation_management/v0/pdf_constructor/base.rb [hunk 1]
```diff
@@ -11,9 +11,9 @@ def initialize(tempfile)
           @template_path = nil
         end
 
-        def construct(data)
+        def construct(data, flatten: true)
```
Comment (holdenhinkle) [inline modules/representation_management/lib/representation_management/v0/pdf_constructor/base.rb:14]: Please add comments about this method (like you added for the other methods) along with a note that flatten is only set to false during tests, and explain why.

File: modules/representation_management/spec/lib/representation_management/v0/pdf_constructor/form_2122_spec.rb [hunk 2]
```diff
@@ -84,11 +105,31 @@
...
+                                     '21-22',
                                      'v0',
+                                     'default',
                                      '2122_with_limitations_no_claimant.pdf')
       expect(tempfile.path).to match_pdf_content_of(expected_pdf)
     end
     # The Tempfile is automatically deleted after the block ends
   end
+
+  it 'matches the field values of a pdf with conditions present and no claimant when unflattened' do
```
Comment (holdenhinkle) [inline modules/representation_management/spec/lib/representation_management/v0/pdf_constructor/form_2122_spec.rb:117]: Nit: perhaps reword for readability?

"matches the field values of a pdf with no claimant with conditions present when unflattened"

File: modules/representation_management/lib/representation_management/v0/pdf_constructor/base.rb (hunk 1 above)
Comment (holdenhinkle) [inline modules/representation_management/lib/representation_management/v0/pdf_constructor/base.rb:14]: Please add comments about this method (like you added for the other methods) along with a note that flatten is only set to false during tests, and explain why.

File: modules/representation_management/spec/lib/representation_management/v0/pdf_constructor/form_2122a_spec.rb [hunk 3]
```diff
@@ -104,9 +125,29 @@
...
                                      '21-22A',
                                      'v0',
+                                     'default',
                                      '2122a_conditions_and_limitations_no_claimant.pdf')
       expect(tempfile.path).to match_pdf_content_of(expected_pdf)
     end
     # The Tempfile is automatically deleted after the block ends
   end
+
+  it 'matches the field values of a pdf with conditions present and no claimant when unflattened' do
```
Comment (holdenhinkle) [inline modules/representation_management/spec/lib/representation_management/v0/pdf_constructor/form_2122a_spec.rb:135]: Same as the previous comment.

**Review by holdenhinkle — COMMENTED** [review]
Looks great, Josh. Just a couple of minor comments.

File: modules/representation_management/spec/lib/representation_management/v0/pdf_constructor/form_2122_spec.rb (hunk 2 above)
Comment (opticbob) [reply modules/representation_management/spec/lib/representation_management/v0/pdf_constructor/form_2122_spec.rb]: Done!

File: modules/representation_management/spec/lib/representation_management/v0/pdf_constructor/form_2122a_spec.rb (hunk 3 above)
Comment (opticbob) [reply modules/representation_management/spec/lib/representation_management/v0/pdf_constructor/form_2122a_spec.rb]: Done!

File: modules/representation_management/lib/representation_management/v0/pdf_constructor/base.rb (hunk 1 above)
Comment (opticbob) [reply modules/representation_management/lib/representation_management/v0/pdf_constructor/base.rb:26]: I added rubydoc annotations to every method where we pass the `flatten` attribute through or use it directly.

**Review by holdenhinkle — APPROVED** [review]
Looks great!

**Review by ericboehs — APPROVED** [review]

**Pull request merged**